framework/web/Request.php

```diff
@@ -187,6 +188,24 @@ public function resolve()
             }
             return [$route, $this->getQueryParams()];
         } else {
+            if (Yii::$app->getUrlManager()->enablePrettyUrl && !empty(Yii::$app->getUrlManager()->suffix)) {
+                $pathInfo = $this->getPathInfo();
+                $suffix = Yii::$app->getUrlManager()->suffix;
+                $n = strlen($suffix);
+                if (substr_compare($pathInfo, $suffix, -$n, $n) !== 0) {
+                    $this->setPathInfo($pathInfo . $suffix);
+                    $result = Yii::$app->getUrlManager()->parseRequest($this);
+                    if ($result !== false) {
+                        $url = $this->getBaseUrl() . '/' . $this->getPathInfo();
```

Comment:
this should be getScriptUrl instead of baseUrl:
$url = $this->getScriptUrl() . '/' . $this->getPathInfo();

Otherwise:
- http://localhost/dmn/yii-application/frontend/web/index.php/site/about will redirect to:
- http://localhost/dmn/yii-application/frontend/web/site/about